resources/windows/wix/service.component.wxs.erb

```diff
@@ -0,0 +1,55 @@
+<?xml version='1.0' encoding='windows-1252'?>
+<Wix xmlns='http://schemas.microsoft.com/wix/2006/wi' xmlns:util="http://schemas.microsoft.com/wix/UtilExtension">
+  <Fragment>
+  <%- get_services.each do |service| -%>
+    <%- service.options.each do |component| -%>
+    <DirectoryRef Id="<%= component[:directory_ref] %>">
+      <Component Id='<%= component[:id] %>' Guid="<%= component[:guid] %>" <%= "Win64=\"#{component[:win64]}\"" unless component[:win64].nil? %>>
+        <%- unless component[:conditions].nil? -%>
+          <%- component[:conditions].each do |condition| -%>
+        <Condition>
+          <%= condition %>
+        </Condition>
+          <%- end -%>
+        <%- end -%>
+        <File Id="<%= component[:file][:id] %>" KeyPath="yes" Source="<%= component[:file][:source] %>" />
+        <ServiceInstall Id="<%= component[:serviceinstall][:id] %>"
+          Account="[PUPPET_AGENT_ACCOUNT_DOMAIN]\[PUPPET_AGENT_ACCOUNT_USER]"
+          Password="[PUPPET_AGENT_ACCOUNT_PASSWORD]"
```

Comment:
A little late to the conversation, but most of this looks good and I wouldn't consider it an abuse of Ruby at all.

These particular bits though, I can possibly foresee folks using different accounts for different services and would want to move to maybe this being more open to that.